### PR TITLE
Create and use qmcpack_complex executable

### PR DESCRIPTION
**Pull request by prckent**
## Proposed changes

To improve overall user experience, when QMC_COMPLEX=1, this PR changes the built executable to  qmcpack_complex. This allows real and complex builds to exist on the same path and moves us closer to a combined userfriendly install ( e.g. https://github.com/QMCPACK/qmcpack/issues/3151 ).

I verified that no occurences of "bin/qmcpack" remain in the configuration files.

## What type(s) of changes does this code introduce?

- New feature
- Build related changes
- Testing changes (e.g. new unit/integration/performance tests)
- Documentation changes

### Does this introduce a breaking change?

- Yes

## What systems has this change been tested on?

nitrogen, nightly configuration, llvm no mpi.

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- NA. Code added or changed in the PR has been clang-formatted
- NA. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

File: tests/performance/C-graphite/CMakeLists.txt [hunk 1]
```diff
@@ -8,8 +8,8 @@ FUNCTION(ADD_C_GRAPHITE_TEST TEST_NAME TEST_DIR INPUT_FILE H5_FILE)
   MESSAGE_VERBOSE("Adding test ${TEST_NAME}")
   SET (WDIR "${qmcpack_BINARY_DIR}/tests/performance/C-graphite/${TEST_DIR}")
   SET (SDIR "${qmcpack_SOURCE_DIR}/tests/performance/C-graphite/sample/${TEST_DIR}")
-
-  SET(QMC_APP "${qmcpack_BINARY_DIR}/bin/qmcpack")
+  GET_TARGET_PROPERTY( QMC_APP_NAME qmcpack OUTPUT_NAME )
+  SET ( QMC_APP "${qmcpack_BINARY_DIR}/bin/${QMC_APP_NAME}" )
```
Comment (ye-luo) [inline tests/performance/C-graphite/CMakeLists.txt:12]: I think "${qmcpack_BINARY_DIR}/bin/${QMC_APP_NAME}" can be replaced with generation expression `$<TARGET_FILE:qmcpack>`

Comment (prckent) [reply tests/performance/C-graphite/CMakeLists.txt:12]: This is working. Not touching it.

Comment (ye-luo) [reply tests/performance/C-graphite/CMakeLists.txt:12]: The requested change works and make the cmake files sane. Unless touching those lines introduce merge conflicts with upcoming PRs, getting them changed in this PR is still preferred.

Comment (prckent) [reply tests/performance/C-graphite/CMakeLists.txt:12]: Investigating

Comment (prckent) [reply tests/performance/C-graphite/CMakeLists.txt:12]: If you want this you'll have to push something that works.

Comment (ye-luo) [reply tests/performance/C-graphite/CMakeLists.txt:12]: I changed macro. Please check if it works on your side, if yes, please propagate the change to other places, if no, revert my commit and I will investigate separately.

**Comment by prckent** [comment]
FYI, what you pushed does not work for me.

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
> FYI, what you pushed does not work for me.
> 
> ```
> cmake -GNinja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DQMC_MPI=0 -DQMC_NIO_MAX_SIZE=64 -DQMC_DATA=/scratch/pk7/QMC_DATA_FULL -DQE_BIN=/scratch/pk7/QMCPACK_CI_BUILDS_DO_NOT_REMOVE/build_gccnew_QE/qe-6.7.0/bin/ -DQMC_COMPLEX=1 ../qmcpack/; ninja; ls -l bin; ctest -j 16 -L deterministic
> ```

I have a fix for this. non MPI case was causing the issue. It seems that I have to push changes into this PR directly otherwise github-action cannot pass.

File: CMake/macros.cmake [hunk 2]
```diff
@@ -108,7 +108,7 @@ FUNCTION( RUN_QMC_APP_NO_COPY TESTNAME WORKDIR PROCS THREADS TEST_ADDED TEST_LAB
         ENDIF()
     ELSE()
         IF ( ( ${PROCS} STREQUAL "1" ) )
-            ADD_TEST( ${TESTNAME} ${QMC_APP} ${ARGN} )
```
Comment (ye-luo) [inline CMake/macros.cmake:111]: @prckent this old signature of ADD_TEST caused generator expression to fail.

**Comment by prckent** [comment]
Confirming both real and complex builds are working for me with the last change. OK to merge this improved version imo.

**Review by ye-luo — APPROVED** [review]

**Pull request merged**